src/Sulu/Component/Content/SmartContent/PageDataProvider.php

```diff
@@ -89,14 +103,18 @@ public function __construct(
         SessionInterface $session,
         ReferenceStoreInterface $referenceStore,
         $showDrafts,
-        $permissions
+        $permissions,
+        FormMetadataProvider $formMetadataProvider = null,
+        TokenStorageInterface $tokenStorage = null
     ) {
         $this->contentQueryBuilder = $contentQueryBuilder;
         $this->contentQueryExecutor = $contentQueryExecutor;
         $this->documentManager = $documentManager;
         $this->proxyFactory = $proxyFactory;
         $this->session = $session;
         $this->referenceStore = $referenceStore;
+        $this->formMetadataProvider = $formMetadataProvider;
```

Comment:
Same as above, it should trigger a deprecation if `$formMetadataProvider` is set to null.